src/cli/CMakeLists.txt

```diff
@@ -27,18 +27,17 @@ list(SORT CLI_SRC_C)
 if(BUILD_CLI STREQUAL "dynamic")
 	set(CLI_LIBGIT2_LIBRARY libgit2package)
 else()
-	set(CLI_LIBGIT2_OBJECTS $<TARGET_OBJECTS:libgit2>)
+	set(CLI_LIBGIT2_LIBRARY libgit2)
 endif()
 
 #
 # Compile and link the CLI
 #
 
 add_executable(git2_cli ${CLI_SRC_C} ${CLI_SRC_OS} ${CLI_OBJECTS}
-	$<TARGET_OBJECTS:util>
 	${CLI_LIBGIT2_OBJECTS}
 	${LIBGIT2_DEPENDENCY_OBJECTS})
-target_link_libraries(git2_cli ${CLI_LIBGIT2_LIBRARY} ${LIBGIT2_SYSTEM_LIBS})
+target_link_libraries(git2_cli ${CLI_LIBGIT2_LIBRARY} ${LIBGIT2_SYSTEM_LIBS} libgit2_xdiff http_parser util)
```

Comment:
I understand why `util` is here, since the CLI uses the utility functions directly. But it doesn't use http parser or xdiff, so I'm surprised that these need to be specified here.